File: spark/src/main/scala/com/adatao/ddf/spark/etl/SparkPersistenceHandler.scala
package com.adatao.ddf.spark.etl

import com.adatao.ddf.DDF
import com.adatao.ddf.spark.{SparkDDFManager}
import com.adatao.ddf.spark.content.SparkRepresentationHandler
import org.apache.spark.rdd.RDD
import com.adatao.ddf.etl.APersistenceHandler
import shark.SharkContext
import shark.api.TableRDD
import shark.memstore2.TablePartition
import com.adatao.ddf.content.{Schema}
import com.adatao.ddf.content.Schema.ColumnType

class SparkPersistenceHandler(container: SparkDDFManager, var sharkContext: SharkContext= null) extends APersistenceHandler(container) {

  def sql2rdd(command: String): TableRDD = {

    sharkContext.sql2rdd(command)
  }

  override def sqlLoad(command: String): Unit = {
    this.resetRepresentationsAndViews()

		// load the table into an RDD, then associate it with the current DDF
		//val rdd: RDD[TablePartition] = null
		val table= this.sql2rdd(command)
    val rdd: RDD[TablePartition] = table.map{
      row => row.rawdata.asInstanceOf[TablePartition]
    }

    table.schema.foreach{
      col => {
        val colType= col.typeName match {
          case "int" => ColumnType.INTEGER
          case "double" => ColumnType.DOUBLE
          case "string" => ColumnType.STRING
        }
        val column= new Schema.Column(col.columnName, colType)
        container.getMetaDataHandler.getSchema.addColumn(column)
      }
    }

		this.getDDF().getDDFManager().getRepresentationHandler().asInstanceOf[SparkRepresentationHandler].set(rdd)
	}

  def sql(command: String): Array[String]= {
    sharkContext.sql(command).toArray
  }
	override def sqlSave(connection: String, command: String): Unit = { null }

	//override def jdbcLoad(connection: String, command: String): DDF = { null }

	//override def jdbcSave(connection: String, command: String): Unit = { null }
}

File: spark/src/main/java/com/adatao/spark/ddf/etl/SparkPersistenceHandler.java
/**
 * 
 */
package com.adatao.spark.ddf.etl;

import org.apache.spark.rdd.RDD;

import shark.SharkContext;
import shark.api.Row;
import shark.api.TableRDD;

import com.adatao.ddf.ADDFManager;
import com.adatao.ddf.DDF;
import com.adatao.ddf.content.Schema;
import com.adatao.ddf.content.Schema.DataFormat;
import com.adatao.ddf.etl.APersistenceHandler;
import com.adatao.ddf.exception.DDFException;
import com.adatao.spark.ddf.SparkDDF;
import com.adatao.spark.ddf.SparkDDFManager;
import com.adatao.spark.ddf.content.SparkSchemaHandler;

/**
 * @author ctn
 * 
 */
public class SparkPersistenceHandler extends APersistenceHandler {

  public SparkPersistenceHandler(ADDFManager theContainer) {
    super(theContainer);
  }

  private SharkContext getSharkContext() {
    return ((SparkDDFManager) this.getContainer()).getSharkContext();
  }

  @Override
  public DDF load(String command) throws DDFException {
    TableRDD tableRdd = this.getSharkContext().sql2rdd(command);
    RDD<Row> rdd = (RDD<Row>) tableRdd;
    Schema schema = SparkSchemaHandler.getSchemaFrom(tableRdd.schema());

    return new SparkDDF(rdd, Row.class, schema);
  }

  @Override
  public DDF load(String command, Schema schema) throws DDFException {
    // TODO Auto-generated method stub
    return null;
  }

  @Override
  public DDF load(String command, DataFormat dataFormat) throws DDFException {
    // TODO Auto-generated method stub
    return null;
  }

  @Override
  public DDF load(String command, Schema schema, String dataSource) throws DDFException {
    // TODO Auto-generated method stub
    return null;
  }

  @Override
  public DDF load(String command, Schema schema, DataFormat dataFormat) throws DDFException {
    // TODO Auto-generated method stub
    return null;
  }

  @Override
  public DDF load(String command, Schema schema, String dataSource, DataFormat dataFormat) throws DDFException {
    // TODO Auto-generated method stub
    return null;
  }
}
